src/server/scripts/Northrend/Ulduar/Ulduar/boss_thorim.cpp

```diff
@@ -66,27 +522,397 @@ class boss_thorim : public CreatureScript
+                            for (Creature* bunny : triggers)
+                                bunny->m_Events.AddEvent(new LightningFieldEvent(bunny), bunny->m_Events.CalculateTime(timer += 100));
+
+                            triggers.remove_if([](Creature* bunny)
+                            {
+                                return LightningFieldCenter.GetExactDist2dSq(bunny) < 400.0f;
+                            });
+
+                            triggers.sort([](Creature* a, Creature* b)
+                            {
+                                return a->GetPositionX() < b->GetPositionX();
+                            });
+
+                            for (std::list<Creature*>::const_iterator itr = triggers.begin(); itr != triggers.end();)
+                            {
+                                std::list<Creature*>::const_iterator prev = itr++;
+                                if (itr != triggers.end())
+                                    (*prev)->CastSpell(*itr, SPELL_LIGHTNING_BEAM_CHANNEL);
+                            }
+                            break;
+                        }
+                        case EVENT_OUTRO_1:
+                            Talk(_hardMode ? SAY_END_HARD_1 : SAY_END_NORMAL_1);
+                            if (_hardMode)
+                                DoCast(me, SPELL_STORMHAMMER_SIF);
+                            break;
+                        case EVENT_OUTRO_2:
+                            Talk(_hardMode ? SAY_END_HARD_2 : SAY_END_NORMAL_2);
+                            if (_hardMode)
+                                if (Creature* Sif = ObjectAccessor::GetCreature(*me, instance->GetGuidData(DATA_SIF)))
+                                    Sif->SetStandState(UNIT_STAND_STATE_DEAD);
+                            break;
+                        case EVENT_OUTRO_3:
+                            Talk(_hardMode ? SAY_END_HARD_3 : SAY_END_NORMAL_3);
+                            break;
+                        default:
+                            break;
+                    }
+                }
+
                 DoMeleeAttackIfReady();
             }
+
+            void DoAction(int32 action) override
+            {
+                switch (action)
+                {
+                    case ACTION_BERSERK:
+                        if (events.IsInPhase(PHASE_2))
+                            return;
+
+                        if (!_orbSummoned)
+                        {
+                            _orbSummoned = true;
+                            events.RescheduleEvent(EVENT_BERSERK, 1000);
+                        }
+                        return;
+                    case ACTION_INCREASE_PREADDS_COUNT:
+                        if (++_killedCount >= 6)
+                        {
+                            // Event starts
+                            me->RemoveFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_IMMUNE_TO_PC);
+                            DoZoneInCombat(me, 250.0f);
```

Comment:
250.0f is new default value for DoZoneInCombat, check : https://github.com/TrinityCore/TrinityCore/blob/3.3.5/src/server/game/AI/CreatureAI.cpp#L49